metrics/group_coordinators.go

```diff
@@ -29,11 +29,11 @@ func NewCoordinatorMetricGroup(info *MetricInfo) *CoordinatorMetricGroup {
 
 func (c *CoordinatorMetricGroup) GetMetrics(status *models.FullStatus) {
 	scope := c.GetScopeOrExit("default")
-	if status == nil || status.Client == nil || status.Client.Coordinators == nil {
+	if !isValidClient(status) || status.Client.Coordinators == nil {
 		log.Error().Msg("failed to get coordinators metric group")
 		return
 	}
-	SetBoolGauge(scope, "quorum", GetBaseTags(), status.Client.Coordinators.QuorumReachable)
+	SetGauge(scope, "quorum", GetBaseTags(), status.Client.Coordinators.QuorumReachable)
 	reachableCount := 0
 	unreachableCount := 0
 	for _, coordinator := range status.Client.Coordinators.Coordinators {
```

Comment:
I think there is not. It can'e be done in a generic way, and that way, we would just split the same code between 2 packages. I would rather have one package to be responsible for the parsing, and the other to convert the parsed data to metrics.